[Term Entry] Python:NumPy built-in-functions Matrix Determinant (#4505)
Codecademy/docs: content/numpy/concepts/built-in-functions/terms/determinant/determinant.md

---
Title: '.linalg.det()' 
Description: 'Returns the determinant of a 2D array' 
Subjects:
  -  Computer Science
  -  Data Science
  -  Machine Learning
Tags:
  -  Linear Algebra
  -  Arrays
  -  NumPy
  -  Functions
CatalogContent: 
  - 'learn-python-3'
  - 'paths/data-science'
---

The **`.linalg.det()`** function returns the determinant of a given matrix. 

## Syntax

```pseudo
numpy.linalg.det(array)
```

`.linalg.det()` provides the following argument:

>  - `array` : The numpy array_like object to compute the determinant for

Reviewer: ```suggestion
- `array` : The numpy array-like object for which the determinant is computed.
```

## Example

> The following example creates multiple numpy arrays then computes and returns the respective determinants.

Reviewer: ```suggestion
The following examples demonstrate the use of `.det()` method:
```

```py
import numpy as np

> a = np.array([[1, 2], [3, 4]])
> det_a = np.linalg.det(a)
> print(det_a)
> 
> b = np.array([[5, 6], [7, 8]])
> det_b = np.linalg.det(b)
> print(det_b)
> 
> c = np.array([[1, 2, 3], [0, 1, 4], [5, 6, 0]])
> det_c = np.linalg.det(c)
> print(det_c)

Reviewer: ```suggestion
mat_2x2 = np.array([[1, 2], [3, 4]])
det_mat_2x2 = np.linalg.det(mat_2x2)
print("Determinant of 2 x 2 matrix:", det_mat_2x2)  # prints output approx. -2.0

mat_3x3 = np.array([[1, 2, 3], [0, 1, 4], [5, 6, 0]])
det_mat_3x3 = np.linalg.det(mat_3x3)
print("Determinant of 3 x 3 matrix:", det_mat_3x3) # prints output approx. 1.0
```

```

This produces the following output:

```shell

> -2.0
> -2.0
> 1.0

Reviewer: ```suggestion
Determinant of 2 x 2 matrix: -2.0000000000000004
Determinant of 3 x 3 matrix: 0.9999999999999964
```

> ```

Reviewer: Please add one codebyte example here.
